code/modules/mining/boulder_processing/_boulder_processing.dm

```diff
@@ -0,0 +1,354 @@
+
+
+/obj/machinery/bouldertech/CanAllowThrough(atom/movable/mover, border_dir)
+	if(!anchored)
+		return FALSE
+	if(boulders_contained.len >= boulders_held_max)
+		return FALSE
+	if(istype(mover, /obj/item/boulder))
+		var/obj/item/boulder/boulder = mover
+		return boulder.can_get_processed()
+	return ..()
+
+/obj/machinery/bouldertech/examine(mob/user)
+	. = ..()
+	if(holds_mining_points)
+		. += span_notice("The machine reads that it has [span_bold("[points_held] mining points")] stored. Swipe an ID to claim them.")
+
+/**
+ * Accepts a boulder into the machinery, then converts it into minerals.
+ * If the boulder can be fully processed by this machine, we take the materials, insert it into the silo, and destroy the boulder.
+ * If the boulder has materials left, we make a copy of the boulder to hold the processable materials, take the processable parts, and eject the original boulder.
+ * @param chosen_boulder The boulder to being breaking down into minerals.
+ */
+/obj/machinery/bouldertech/proc/breakdown_boulder(obj/item/boulder/chosen_boulder)
+	if(QDELETED(chosen_boulder))
+		return FALSE
+	if(chosen_boulder.loc != src)
+		return FALSE
+	if(!length(chosen_boulder.custom_materials))
+		qdel(chosen_boulder)
+		playsound(loc, 'sound/weapons/drill.ogg', 50, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+		update_boulder_count()
+		return FALSE
+	if(isnull(silo_materials))
+		return FALSE
+
+	//here we loop through the boulder's ores
+	var/list/processable_ores = list()
+	var/tripped = FALSE
+	//If a material is in the boulder's custom_materials, but not in the processable_materials list, we add it to the processable_ores list to add back to a leftover boulder.
+	for(var/datum/material/possible_mat as anything in chosen_boulder.custom_materials)
+		if(!is_type_in_list(possible_mat, processable_materials))
+			continue
+			var/quantity = chosen_boulder.custom_materials[possible_mat]
+			points_held = round(points_held + (quantity * possible_mat.points_per_unit)) // put point total here into machine
+			processable_ores += possible_mat
+			processable_ores[possible_mat] = quantity
+			chosen_boulder.custom_materials -= possible_mat //Remove it from the boulder now that it's tracked
+			tripped = TRUE
+	if(!tripped)
+		remove_boulder(chosen_boulder)
+		say("Nothing to process!")
+		return FALSE //we shouldn't spend more time processing a boulder with contents we don't care about.
+	use_power(BASE_MACHINE_ACTIVE_CONSUMPTION)
+	check_for_boosts() //Calls the relevant behavior for boosting the machine's efficiency, if able.
+	var/is_artifact = (istype(chosen_boulder, /obj/item/boulder/artifact)) //We need to know if it's an artifact so we can carry it over to the new boulder.
+	var/obj/item/boulder/disposable_boulder = new (src)
+	disposable_boulder.custom_materials = processable_ores
+	silo_materials.mat_container.insert_item(
+		weapon = disposable_boulder,\
+		multiplier = refining_efficiency,\
+		breakdown_flags = BREAKDOWN_FLAGS_ORM,\
+		)
```

Comment:
I was wrong about my review for `context`. You do need to add `context = src` as the 4th param

Reply:
Change of review. I have made an helper proc that makes inserting easy without needing to specify context.

Try using the insert proc defined on remote materials directly and not in material container